### PR TITLE
fix(snowflake): deprecate config, update examples

### PR DESCRIPTION
**Pull request by anshbansal**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [x] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source_config/sql/snowflake.py [hunk 1]
```diff
@@ -75,20 +79,48 @@ class BaseSnowflakeConfig(BaseTimeWindowConfig):
...
+        if host_port is not None:
+            logger.warning(
+                "snowflake's `host_port` option has been deprecated; use account_id instead"
+            )
+            host_port = cls._clean_account(v=host_port)
+            values["host_port"] = host_port
+        account_id = values.get("account_id")
+        if account_id is None:
+            if host_port is None:
+                raise ConfigurationError(
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/ingestion/source_config/sql/snowflake.py:116]: Please add test coverage for this case.

File: metadata-ingestion/src/datahub/ingestion/source_config/sql/snowflake.py [hunk 2]
```diff
@@ -75,20 +79,48 @@ class BaseSnowflakeConfig(BaseTimeWindowConfig):
...
         v = remove_protocol(v)
         v = remove_trailing_slashes(v)
         v = remove_suffix(v, ".snowflakecomputing.com")
         return v
 
+    @pydantic.root_validator
+    def one_of_host_port_or_account_id_is_required(cls, values):
+        host_port = values.get("host_port")
+        if host_port is not None:
+            logger.warning(
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/ingestion/source_config/sql/snowflake.py:108]: Is this case being covered in the tests?

File: metadata-ingestion/src/datahub/ingestion/source_config/sql/snowflake.py (hunk 1 above)
Comment (anshbansal) [reply metadata-ingestion/src/datahub/ingestion/source_config/sql/snowflake.py:116]: added

File: metadata-ingestion/src/datahub/ingestion/source_config/sql/snowflake.py (hunk 2 above)
Comment (anshbansal) [reply metadata-ingestion/src/datahub/ingestion/source_config/sql/snowflake.py:108]: added

**Review by rslanka — APPROVED** [review]
LGTM!

**Pull request merged**